test/root/Drainable.test.js

```diff
@@ -0,0 +1,80 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+
+import deployer from '../helpers/deployer.js'
+import * as utils from '../helpers/utils'
+
+chai
+  .use(chaiAsPromised)
+  .should()
+
+contract('Drainable', async function(accounts) {
+  let depositManager, drainable, destination
+  const amount = web3.utils.toBN('1').pow(web3.utils.toBN('18'))
+  describe('drain assets from Deposit Manager', async function() {
+    before(async function() {
+      this.contracts = await deployer.freshDeploy()
+    })
+    beforeEach(async function() {
+      await deployer.deployRootChain()
+      depositManager = await deployer.deployDepositManager()
+      let randomUser = await web3.eth.accounts.create()
+      destination = randomUser.address
+    })
+
+    it('drain ERC20', async function() {
+      const testToken = await deployer.deployTestErc20()
+      await testToken.transfer(depositManager.address, amount)
+      utils.assertBigNumberEquality(await testToken.balanceOf(depositManager.address), amount)
+      const tokens = []
+      const values = []
+      tokens.push(testToken.address)
+      values.push(amount.toString())
+      drainable = await deployer.deployDrainable()
+      const data = drainable.contract.methods.drainErc20(tokens, values, destination).encodeABI()
+      await this.contracts.governance.update(
+        drainable.address,
+        data
+      )
+      utils.assertBigNumberEquality(await testToken.balanceOf(destination), amount)
```

Comment:
Did you remove the assertion where balanceOf deposit Manager should now be 0. Also can we test for partial amount? Like if DepositManager has 10 we should drain only 7.

Reply:
yes, was that required? assertion for the user to get the queried amount of tokens should be enough, no?

Reply:
yes, will add for partial amount too

Reply:
> assertion for the user to get the queried amount of tokens should be enough, no
Isn't that just half the assertion? Better to write a complete story.